ssl/ssl_asn1.c

```diff
@@ -229,7 +229,7 @@ static int ssl_session_strndup(char **pdst, ASN1_OCTET_STRING *src)
 static int ssl_session_memcpy(unsigned char *dst, size_t *pdstlen,
                               ASN1_OCTET_STRING *src, size_t maxlen)
 {
-    if (src == NULL) {
+    if (src == NULL || src->length == 0) {
```

Comment:
Here as well. memcpy of 0 bytes is a well-defined, only memcpy 0 bytes from NULL is to avoid.

Reply:
And neither can this. I am keeping it.